### ISSUE TITLE
Add support for calculating a median score for audits?

### ISSUE THREAD

**Issue by addyosmani** [GoogleChrome/lighthouse#792]
**tl;dr: should we add support for multiple runs?** 🙃 

It's not uncommon to see variances in the scores of performance audits such as the PSI, FMP, EIL and TTI. Sometimes variances are larger than others 🙈 

Here's an example with the latest version of LH on housing.com across 3 runs:

Run 1:
![](https://i.imgur.com/dTPSWsC.jpg)

Run 2:
![](https://i.imgur.com/6nse6Lw.jpg)

Run 3:
![](https://i.imgur.com/b9q2ko8.jpg)
- The time to interactive ranges from 700ms to 1626ms. **Median: 741**
- First meaningful paint ranges from 688 to 1365: **Median: 730**

On my MBP, it takes Lighthouse about this much ⏲  to do a run of the above site:

```
real    0m18.604s
user    0m15.009s
```

With that in mind, I don't think it makes sense to introduce multi-run medians as a default. But for folks that don't mind the wait, perhaps we could support a `--runs=N` flag or similar that takes the aggregate scores and provides a report with the medians in there?

I was about to build a wrapper around Lighthouse to do this but was unsure if this type of thing belonged in core or not.

**Comment by brendankenny** [comment]
One of our goals this quarter is to better characterize our performance tests, being able to make confident statements about results and if a run shows an improvement or not over previous runs. I think this would be a great first step.

I'm personally fine with adding it as a flag, but putting it into the report requires touching the increasingly creaky scoring code, so it will probably be a lot simpler to write a quick wrapper utility for now that we can adopt into core after we've updated scoring for #746, etc

**Comment by addyosmani** [comment]
> it will probably be a lot simpler to write a quick wrapper utility for now that we can adopt into core after we've updated scoring for #746, etc

This sounds like a fair short term solution to me 👍

**Comment by brendankenny** [comment]
sidenote: the Speed Index for that last run seems really, *really* wrong given the other perf numbers (first visual change at 50s while FMP was at 730ms?). Definitely a bug :)

**Comment by addyosmani** [comment]
> sidenote: the Speed Index for that last run seems really, really wrong given the other perf numbers (first visual change at 50s while FMP was at 730ms?). Definitely a bug :)

I figured the PSI for the same runs wouldn't reflect the same issue but perhaps that's being overly optimistic :) I'll do a double check.

**Comment by patrickhulce** [comment]
Seems decidedly deferred to projects like pwmetrics at this point?
